src/overlays/actors/ovl_En_Nutsball/z_en_nutsball.c

```diff
@@ -21,14 +23,146 @@ const ActorInit En_Nutsball_InitVars = {
     (ActorFunc)EnNutsball_Update,
     (ActorFunc)EnNutsball_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Nutsball_0x80985C40/EnNutsball_Init.asm")
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_NONE, 
+        AT_ON | AT_TYPE_ENEMY, 
+        AC_ON | AC_TYPE_PLAYER, 
+        OC1_ON | OC1_TYPE_ALL, 
+        OC2_TYPE_2, 
+        COLSHAPE_CYLINDER, 
+    },
+    { 
+        ELEMTYPE_UNK0, 
+        { 0xF7CFFFFF, 0x00, 0x04 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_ON | TOUCH_SFX_WOOD, 
+        BUMP_ON, 
+        OCELEM_ON, 
+    },
+    { 13, 13, 0, { 0, 0, 0 } },
+};
+
+void EnNutsball_Init(Actor *thisx, GlobalContext *globalCtx) {
+    EnNutsball *this = THIS;
+    
+    ActorShape_Init(&this->actor.shape, 400.0f, (ActorShadowFunc)func_800B3FC0, 13.0f);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.shape.rot.y = 0;
+    this->actor.speedXZ = 10.0f;
+    if (this->actor.params == 2) {
+        this->timer = 1;
+        this->unk146 = 0;
+        this->actor.gravity = -1.0f;
+    } else {
+        this->timer = 20;
+        this->unk146 = 19;
+        this->actor.gravity = -0.5f;
+    }
+    this->actor.world.rot.x = -this->actor.shape.rot.x;
+    this->actor.shape.rot.x = 0;
+    if (this->actor.params == 1) {
+        func_80985D3C(this);
+    }
+}
+
+void EnNutsball_Destroy(Actor *thisx, GlobalContext *globalCtx) {
+    EnNutsball *this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80985D3C(EnNutsball *this) {
+    this->collider.base.atFlags &= ~AT_TYPE_ENEMY & ~AT_BOUNCED & ~AT_HIT;
```

Comment:
Any good name suggestions for this function?

Reply:
`EnNutsball_InitColliderParams`